File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/user/NameCacheService.java
package top.likoslupus.cellulosesz.api.user;

import java.util.Map;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface NameCacheService {

    void remember(UUID uuid, String name);

    Optional<UUID> findUuid(String name);

    Optional<String> findName(UUID uuid);

    Map<UUID, String> entries();

    CompletableFuture<Void> save();

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/user/NameCacheService.kt
package top.likoslupus.cellulosesz.api.user

import java.util.*
import java.util.concurrent.CompletableFuture

public interface NameCacheService {

    public fun remember(uuid: UUID, name: String)

    public fun findUuid(name: String): UUID?

    public fun findName(uuid: UUID): String?

    public fun entries(): Map<UUID, String>

    public fun save(): CompletableFuture<Void>

}
